src/main/java/juja/sqlcmd/DatabaseManager.java

```diff
@@ -5,16 +5,33 @@
 import java.sql.SQLException;
 
 public class DatabaseManager {
+    private final String jdbcDriverClass;
+    private final String jdbcUrl;
     private Connection connection;
 
+    public DatabaseManager() {
+        jdbcDriverClass = "org.postgresql.Driver";
+        jdbcUrl = "jdbc:postgresql://localhost:5432/";
+    }
+
+    public DatabaseManager(String jdbcDriverClass, String jdbcUrl) {
+        this.jdbcDriverClass = jdbcDriverClass;
+        this.jdbcUrl = jdbcUrl;
+    }
+
     boolean connect(String database, String user, String password) {
         try {
-            Class.forName("org.postgresql.Driver");
+            Class.forName(jdbcDriverClass);
+            if (connection != null) {
+                connection.close();
+            }
+
             connection = DriverManager.getConnection(
-                    "jdbc:postgresql://localhost:5432" + database, user, password);
+                    jdbcUrl + database, user, password);
             return true;
         } catch (ClassNotFoundException | SQLException e) {
             e.printStackTrace();
+            connection = null;
```

Comment:
What if connection is not closed at this moment?

Reply:
Good question, I just copied it from Baglai's example)